packages/webdriverio/src/utils/findStrategy.js

```diff
@@ -24,9 +24,11 @@ const defineStrategy = function (selector) {
     // Condition with checking isPlainObject(selector) should be first because
     // in case of "selector" argument is a plain object then .match() will cause
     // an error like "selector.match is not a function"
-    // Use '-android datamatcher' strategy if selector is a plain object (Android only)
+    // Use '-android datamatcher' or '-android viewmatcher' strategy if selector is a plain object (Android only)
     if (isPlainObject(selector)) {
-        return '-android datamatcher'
+        if (JSON.stringify(selector).indexOf('test.espresso.matcher.ViewMatchers') < 0)
```

Comment:
How about:

```suggestion
        if (selector.class === 'test.espresso.matcher.ViewMatchers')
```

Reply:
The ViewMatchers class is not guaranteed to appear in the root object, it can appear in the nested object as well. For ex.
`
{
  name: 'allOf',
  args: [{
     name: 'withId',
     args: 'testId',
     class: 'androidx.test.espresso.matcher.ViewMatchers'
  },
  {
     name: 'withText',
     args: 'testText',
     class: 'androidx.test.espresso.matcher.ViewMatchers'
  }],
  class: 'org.hamcrest.Matcher'
}
`

In Java Espresso - onView(allOf(withId(R.id.my_view), withText("Hello!")));

Reply:
Is there a possibility to have e.g.:
```json
{
   "name":"allOf",
   "args":[
      {
         "name":"withId",
         "args":"testId",
         "class":"androidx.test.espresso.matcher.ViewMatchers"
      },
      {
         "name":"withText",
         "args":"testText",
         "class":"androidx.test.espresso.matcher.DataMatchers"
      }
   ],
   "class":"org.hamcrest.Matcher"
}
```
_Note:_ there is a data and view matcher.

Reply:
Yes, this is also possible. Well...if what you mean is different classes in the args (I don't think that DataMatchers class exist). Whether or not it's going to find the element is another story.